packages/runtime-core/src/errorHandling.ts

```diff
@@ -71,8 +72,8 @@ export function callWithAsyncErrorHandling(
   args?: any[]
 ) {
   const res = callWithErrorHandling(fn, instance, type, args)
-  if (res != null && !res._isVue && typeof res.then === 'function') {
-    res.catch((err: any) => {
+  if (res != null && !res._isVue && isPromise(res)) {
+    res.catch((err: Error) => {
```

Comment:
might be better to use `unknown` on error
src: https://twitter.com/OliverJAsh/status/1184505365012070400?s=20 and the link in the comment

Reply:
yea, and we should also add a type check so only errors will go into handling fn, since it accepts only errors.